resources/assets/lib/interfaces/beatmapset-discussion-post-json.ts

```diff
@@ -14,11 +14,22 @@ interface BeatmapsetDiscussionPostDefaultAttributes {
   deleted_by_id?: number;
   id: number;
   last_editor_id?: number;
-  message: string;
-  system: boolean;
   updated_at: string;
   user_id: number;
 }
 
-type BeatmapsetDiscussionPostJson = BeatmapsetDiscussionPostDefaultAttributes & Partial<BeatmapsetDiscussionPostAvailableIncludes>;
+type BeatmapsetDiscussionMessagePostJson = BeatmapsetDiscussionPostDefaultAttributes & {
+  message: string;
+  system: false;
+};
+
+type BeatmapsetDiscussionSystemPostJson = BeatmapsetDiscussionPostDefaultAttributes & {
+  message: {
+    type: string;
+    value: boolean;
```

Comment:
continuing the stronger typing for `type` field above, `boolean` is probably only applicable for `type: 'resolved'` anyway...

Reply:
there's not but there's no telling what type the value would be in other cases 🤔

Reply:
turns out `{ type: 'resolved'; value: boolean } | { type: string }` is not a thing (always resolves to `string`)

Reply:
...of course it's not. The typing wouldn't make sense either (like, what is it typing?). I think it should just be `{ type: 'resolved'; value: boolean }` for now.

Reply:
that barely changes anything

Reply:
that's because there's only one at the moment but it will allow making sure nothing is missed if more type is added, similar to [user event](https://github.com/ppy/osu-web/blob/d56f0371595d07daf953d2929335017132b162b2/resources/assets/lib/profile-page/parse-event.tsx)